<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.7">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-02-03">

<title>Bioconductor Slack channels</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="BiocSlack_files/libs/clipboard/clipboard.min.js"></script>
<script src="BiocSlack_files/libs/quarto-html/quarto.js"></script>
<script src="BiocSlack_files/libs/quarto-html/popper.min.js"></script>
<script src="BiocSlack_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="BiocSlack_files/libs/quarto-html/anchor.min.js"></script>
<link href="BiocSlack_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="BiocSlack_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="BiocSlack_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="BiocSlack_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="BiocSlack_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="BiocSlack_files/libs/core-js-2.5.3/shim.min.js"></script>
<script src="BiocSlack_files/libs/react-18.2.0/react.min.js"></script>
<script src="BiocSlack_files/libs/react-18.2.0/react-dom.min.js"></script>
<script src="BiocSlack_files/libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="BiocSlack_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="BiocSlack_files/libs/reactable-0.4.4/reactable.css" rel="stylesheet">
<script src="BiocSlack_files/libs/reactable-binding-0.4.4/reactable.js"></script>


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Bioconductor Slack channels</h1>
<p class="subtitle lead">Public channels in <a href="https://slack.bioconductor.org/">https://slack.bioconductor.org/</a>. Information in table updates daily.</p>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 3, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="cell">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-5faf8969494a0cc4d2db" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5faf8969494a0cc4d2db">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"name":["random","bigdata-rep","general","mae-discussion","introductions","new_packages","package_submissions","microbiome_metagenome","hca_rfa","singlecellexperiment","bioc_git","pharmacogenomics","tree-like-se","isee","rhdf5client","meetups","workflows","hca_clustering","delayed_array","rowdatafix","biocverse","palmtree","nullranges","sticker_joy","biochubs","sc-batch-correction","salmon2bioc","sc-signature","gseabase","bioc-builds","tidiness_in_bioc","anvil","containers","biocasia","biocworkshops","papersandpreprints","diversebioc","tech-advisory-board","developers-forum","osca-book","scalability","jobs","support-site","sc-repertoires","bioc-conference-web","bioc-conference-noncompeting","cool-vis","spatial","machine-learning","bioc-conference-sponsorship","biocasia2020","cyto-spec-book","help-","education-and-training","omicidx","covid19","proteomics","metabolomics","community-advisory-board","singlecell-queries","bioc2020-bof-10sr","accessible-vis","rcwl","reproducibility","clinical","sc-differential-expression","sc-velocity","miaverse","tab_polls","bioc_africa","database","eurobioc2020-isee-transcripts","biocchallenges","gwas-data","deedee","humble-brag","microbiome_experiment_bioc2021","r7","biocasia2021","package-submission","biocwebsite","beginners_and_schools","documentation-translation","mass-spectrometry","developers-mentorship","reviewers","miaverse-admins","miaverse-admins","github-actions","nyc2021a","shiny","hca-data-insights","microsoft-azure-collaboration","annotation-roadmap","calendar-events","unitat-de-nutrició-humana","mentorship_program_microstasis","outreachy","oma-developers","sweave2rmd","cancer-related","cpp","zarr","queer-bioc","r-ladies","industry","msmb-quarto-conversion","bioc2023","funding","bioc-connect","social-media","biocasia2022","carpentries","image-analysis","arm-linux","elixir","hacktoberfest","cloud-working-group","biocclasses","bugsigdb","longread","ismb-osca","biocasia2023","csama2023","spatialdata-devel","biocpython","gpu","artifactdb","spatial-interoperability","biocbooks","easylift","cloud-github-actions-subgroup","easyplot","eurobioc-conference-everyone","alpha-missense","other-distros","bioc2u","popgene","wasm","voyager","osca-book-club"],"topic":["Non-work banter and water cooler conversation","Off-disk/on-disk representations in Bioconductor","Link to join the slack team - <https://slack.bioconductor.org/>","Discuss the MultiAssayExperiment","","","","","","","","","Data Structure","Interactive SummarizedExperiment Explorer","","","","","","","","Visualization","Development and support channel for {nullranges} and associated packages","","","","","Extraction and application of cell \"type\" signatures","","","","<https://meet.google.com/ied-ouvi-sey>","","","Building Bioconductor workshops","","Discussion of diversity and inclusiveness of the Bioconductor community.","","Platform for BioC developers to describe existing software infrastructure to other members of the BioC community, to present plans for future developments, and discuss changes that may impact other developers/software tools within the BioC.","","","","You can report issues at <https://github.com/Bioconductor/support.bioconductor.org/issues>","","Development of the website <http://bioc2022.bioconductor.org|bioc2022.bioconductor.org>, see <https://github.com/Bioconductor/BioC2022>","Planning for the “non-competing components” part of the Bioc conference. For general conference discussion during the conference use <#CLAEUFVAA|bioc2020> and for workshop-specific discussion use <#C015QR5FCGN|bioc2020-workshops>.","","discuss data infrastructure and analysis methods for spatial data in Bioconductor","","For sponsorship-related discussions during planning of BioC2020. For general conference discussion during the conference use <#CLAEUFVAA|bioc2020> and for workshop-specific discussion use <#C015QR5FCGN|bioc2020-workshops>.","","","","Post your own materials, discuss processes, announce courses, and perhaps do some collaborative curriculum/material development","","","","","Public Community Advisory Board channel: it’s a bridge between the CAB and the rest of the community :smiley:","Please note: Seurat is not a bioconductor package and its authors are not here","","","","","","","","Microbiomes with SummarizedExperiment: <https://microbiome.github.io>","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","Group for Bioconductor users and developers who work in industry or non academic settings. Aligned to industry working group ","","","","","","","","","","","","","","","","","","only for those who like chatting on Slack — use Google Docs and email for important communication","","","","","","","","","","EuroBioC2023 conference.","","","","","","",""],"purpose":["A place for non-work-related flimflam, faffing, hodge-podge or jibber-jabber you'd prefer to keep out of more focused work-related channels.","","This channel is for team-wide communication and announcements. All team members are in this channel.","","For members to introduce themselves as they join slack","RSS feed of new packages","","Discussion of metagenomics and microbiome research and links, data and software resources","To coordinate BioC-related submissions to the CZI HCA RFA","Discuss implementation of the SingleCellExperiment class","Discuss care and feeding of git in the context of Bioc","Discuss pharmacogenomics approaches, with a focus on machine learning, drug response and target prediction, and repositioning","Discuss tree-like and hierarchical 'rows' for SummarizedExperiment","Interactive SummarizedExperiment Explorer","discuss R interface to HDF Server and HSDS","Discuss R/Bioconductor meetups and share materials among groups","To talk about the BioC workflows","Discussion on clustering implementations for the CZI-HCA Bioconductor award","","","Create a resource to visualize the Bioconductor ecosystem","","{nullranges} is a modular package to generate feature sets representing the null hypothesis (either through matching on covariates or block bootstrapping of features)","","give feedback or report issues on Bioconductor Hubs","A package for single-cell batch correction methods","Import alevin quantification into Bioc and making nice summary report","Extraction and application of cell \"type\" signatures","Expanding the `GSEABase` package for gene sets and \"signatures\"","A safe space for dealing with BioC build system pain. Suffering shared is suffering halved.","discuss relationship of Bioc structures to tidyverse","Discussions related to the anvil project","discuss container discipline and impacts on maintenance","Chat about &amp; plan the event","","Post your Bioconductor-related manuscripts and preprints here","DiverseBioc : Discussion of diversity in Bioconductor","Discuss topics and activities of the Bioconductor technical advisory board","","To discuss questions, problems, and contributions to the Orchestrating Single Cell Analysis (<https://osca.bioconductor.org>) bookdown book","Focus discussion on scalable computing in Bioc","Bioc related Jobs notice board","Bug reports and feature requests for updated support site","Discussion of single-cell repertoire sequencing","coordinate the efforts for the web page for the bioc conference. Also see: #bioc2022 <#G01KMRQP3C7|bioc-conference-organization> <#CJDMYKG2U|biocworkshops>","Plan the “non-competing components” part of the Bioc conference","Truth and beauty. But mostly the latter.","Discuss data infrastructure and analysis methods for spatial data in Bioconductor","Bioconductor-centric Bioinformatics-flavoured Machine Learning","For sponsorship-related discussions during planning of BioC2021 For general conference discussion during the conference use <#C01GLGEQAGG|bioc2021>","BiocAsia2020, Oct. 15-18, Beijing","","","Discuss all things Bioc education and training-related, network with folks interested","<https://omicidx.cancerdatasci.org/>","Discuss software/data related to COVID-19","Discuss all things bioconductor proteomics related","discussion surrounding metabolomics analyses","Discuss topics and activities of Bioconductor's Community Advisory Board (CAB)","Discuss general queries regarding Single Cell Data Analysis","A place for stimulating discussion about the BioC2020 Birds of a Feather “Ten simple rules for thriving in bioinformatics research”","","The Bioconductor toolset for usage and development of reproducible bioinformatics pipelines in CWL. This is intended to be a community efforts and we welcome collaborations and contributions from the community. Look forward to working with you!","Solutions for reproducible workflows and environments","Discuss clinical applications and standards","Methods, models and tests for sc differential expression analysis","Discussions about the analysis of expression dynamics in single cell RNA seq data","Development of microbiome packages","polling for TAB discussion","Bioconductor in and related to Africa","Working with databases in R","Unlock iSee for transcript-level visualizations","General coordination of BiocChallenges","Continues a discussion of data structures for GWAS-related data, especially summary statistics","","Announce your contributions, awards, grants, talks, achievements,  “today I learnt”, link to your blog, package, gist, self-promotion is welcome here!  Bioconductor  “greatest hit”,  achievements. We want to celebrate your baby steps and giant leaps","Planning channel for Bioc2021","Potential adoption of R7 within Bioconductor","Organising BioC Asia 2021","Ask any question about the submission and review process of new packages here. Packages currently under review are listed here: <https://github.com/Bioconductor/Contributions/issues>","Talk about renewal plan of <http://bioconductor.org|bioconductor.org>","This channel is a place to provide and discuss resources for teaching R and Bioconductor to high school, for undergraduate summer projects, or beginner resources to help welcome non-statistical/computational collaborators to R and Bioconductor","Discuss topics related to translating package documentation","Mass spectrometry data representation and analysis with Bioconductor. See also dedicated #metabolomics and #proteomics channels for more specific questions.","Bioconductor mentorship program to assist new developers and would-be developers to develop a new package. Feel free to talk both the mentorship program itself and to discuss package development.","Discussion channel for active community package reviewers. Ask questions, post issues, ask to swap issues, etc.  Maintainers wishing to ask questions about the package review process or submission process please use package-submission channel.","Miaverse admin channel","Miaverse admin channel","Questions and advice for running GitHub actions for Bioconductor packages","mon sep 20 meeting in nyc","Discussion around use of Shiny within Bioconductor packages!","Human Cell Atlas ‘Data Insights’ proposal","Open channel to talk about the collaboration between Bioconductor and Microsoft Azure","Bioc annotation resources are a central source of project value.  How should this component evolve?","This channel collects bioc-related event information around the world and shares it on Google calendar.","","Channel to communicate about the first Bioconductor mentorship program cycle. Mentee is Pedro Sánchez-Sánchez and the project could be named as microSTASIS","For outreachy applicants to introduce themselves and ask questions.\n\nThe two project repositories for the Dec 22 cohort are\n* <https://github.com/Bioconductor/BSgenomeForge>\n* <https://github.com/Bioconductor/sweave2rmd>","Orchestrating Microbiome Analysis online book development channel","A project with the goal to convert all Bioconductor package vignettes to R Markdown.","Discuss cancer-related data, resources, tools, algorithms, and opportunities","R and C++","Working with data in zarr format in R","","Channel for Bioconductor users and developers who are R-Ladies.","Group for Bioconductor users and developers who work in industry or non academic settings. Aligned to industry working group","","Details on BioC23","Funding opportunities","To identify automated ways to help Bioc people, groups, funders find connections between each other e.g. using IAM, ORCID etc. Benefits are to make it easier to network, identify shared resources/challenges/funding etc","Group for Bioconductor social media discussions (Blog, Twitter, Mastodon, LinkedIn, YouTube, Bilibili, WeChat, Weibo etc). Aligned to social media working group.","Discussions during and about BioC Asia 2022","A channel for the trainees and trainers in the Bioconductor Carpentries training program","Working with biomedical image data in R/Bioc","How can bioc support arm-linux","Connections between Bioconductor and the EU elixir bioinformatics infrastructure network","A channel for discussions of Bioconductor participation in Hacktoberfest","Meeting March 21 @ 11 AM ET : <https://meet.google.com/epw-wems-crb>","Bioconductor Classes Working Group","Discussion related to <http://bugsigdb.org|bugsigdb.org>, the curation of microbial signatures, and the analysis of microbial signature data","Discussions about long-read RNA-seq in Bioconductor and beyond","OSCA tutorial at the ISMB 2023","Organisation of Bioconductor Asia 2023","","S4Class to read/write/represent\nOME-Zarr/SpatialData-Python ","","","Deploying an ArtifactDB instance for Bioconductor","","","","Channel for the GitHub Actions Subgroup (part of the Cloud Methods Working Group).","","This channel is for all participants, speakers and organisers.","","This channel is created to discuss the fact that Bioconductor software is redistributed by a number of projects including macports and conda.  Sometimes the assets shipped are not the most appropriate ones.  How can we enhance user experiences?","","talk the theory, methods, founds in popgene","","",""],"no. members":[2955,280,2984,24,2864,86,34,81,53,407,155,58,48,85,31,99,128,30,54,8,27,17,25,26,76,115,32,105,57,156,159,56,125,30,170,127,102,62,279,116,49,391,33,37,29,21,136,194,137,31,47,21,23,373,4,36,69,52,76,204,92,43,19,68,46,48,32,97,7,31,15,8,11,33,2,78,13,19,17,139,63,39,28,17,46,29,6,6,61,3,67,38,11,29,16,2,5,122,9,57,16,11,40,14,82,62,0,161,60,8,33,45,27,72,13,64,16,44,23,203,23,10,12,14,12,47,0,19,21,18,2,16,3,150,11,3,3,1,13,16,11],"created":["2016-11-21","2016-11-21","2016-11-21","2016-11-21","2016-11-23","2016-12-12","2016-12-14","2017-05-15","2017-07-27","2017-08-08","2017-08-12","2017-09-01","2017-10-19","2017-12-08","2018-02-08","2018-02-26","2018-04-03","2018-04-18","2018-07-09","2018-07-25","2018-07-25","2018-07-26","2018-08-14","2018-10-26","2018-10-31","2018-11-05","2018-11-05","2018-11-20","2018-12-04","2018-12-11","2018-12-10","2018-12-17","2019-03-27","2019-04-02","2019-05-03","2019-05-20","2019-06-28","2019-07-15","2019-08-08","2019-08-15","2019-08-11","2019-08-14","2019-08-30","2019-09-12","2019-10-10","2019-10-11","2019-10-29","2019-11-27","2019-11-29","2019-12-12","2020-01-09","2020-01-29","2020-02-05","2020-03-05","2020-03-11","2020-03-20","2020-03-23","2020-03-23","2020-03-30","2020-05-03","2020-07-29","2020-07-30","2020-07-29","2020-07-21","2020-09-01","2020-09-03","2020-09-05","2020-09-24","2020-10-01","2020-10-20","2020-11-04","2020-12-15","2020-12-18","2021-01-23","2021-03-05","2021-03-03","2021-03-04","2021-03-18","2021-03-16","2021-04-28","2021-04-29","2021-05-07","2021-06-16","2021-07-29","2021-08-05","2021-08-03","2021-08-15","2021-08-15","2021-08-31","2021-09-19","2021-10-07","2021-10-24","2022-02-04","2022-02-06","2022-02-12","2022-02-17","2022-03-02","2022-03-18","2022-04-13","2022-05-09","2022-05-10","2022-05-18","2022-06-28","2022-07-21","2022-07-28","2022-08-12","2022-08-18","2022-09-07","2022-09-16","2022-10-13","2022-11-16","2022-11-28","2022-12-12","2022-12-12","2023-01-05","2023-01-12","2023-01-20","2023-02-09","2023-02-10","2023-02-21","2023-03-03","2023-03-13","2023-04-19","2023-04-19","2023-04-28","2023-05-05","2023-05-05","2023-05-12","2023-06-07","2023-07-06","2023-07-26","2023-07-31","2023-08-29","2023-09-12","2023-09-20","2023-10-30","2023-12-02","2024-01-27","2024-02-01","2022-11-02","2023-07-04"],"latest message":["2024-02-02","2023-01-17","2024-02-01","2021-06-30","2024-01-21","2021-05-13","2024-02-01","2023-12-21","2019-07-01","2023-11-02","2023-11-16","2019-04-23","2023-09-22","2023-11-28","2018-04-27","2024-02-02","2023-09-12","2021-09-14","2023-08-07","2018-07-25","2019-07-08","2019-02-12","2023-06-02","2023-07-26","2023-07-17","2022-02-08","2023-12-15","2023-01-10","2020-11-11","2024-02-01","2024-01-17","2022-12-12","2024-01-12","2023-09-09","2023-07-31","2023-10-18","2023-03-08","2024-02-01","2024-01-31","2024-02-02","2023-04-11","2024-01-24","2023-11-13","2020-05-08","2022-07-29","2020-02-10","2023-12-01","2024-02-03","2024-01-26","2023-05-09","2021-05-30","2020-11-13","2020-05-07","2024-01-26",null,"2020-08-31","2023-06-30","2024-01-09","2024-01-12","2024-02-02","2023-11-09","2023-10-16","2021-04-12","2020-10-30","2023-08-08",null,"2022-01-14","2024-01-27","2020-10-01","2022-05-06","2020-11-04","2021-07-14","2020-12-18","2024-01-26",null,"2023-05-04","2021-08-03","2023-08-25","2022-02-10","2024-01-29","2024-01-29","2023-06-13","2023-12-29","2023-07-11","2023-10-24","2024-01-22","2024-02-01","2024-02-01","2023-04-21",null,"2023-11-09","2023-11-15",null,"2022-03-21","2022-08-31",null,"2022-04-11","2023-10-23","2023-01-28","2024-01-14","2022-05-10","2023-02-23","2023-02-09","2022-08-12","2023-07-13","2023-08-04",null,"2023-09-07","2023-09-06","2022-10-23","2024-01-11","2023-04-28","2023-12-21","2023-08-11","2024-02-01","2023-12-30","2023-09-14","2024-01-23","2023-12-30","2024-02-02","2023-10-14","2023-10-30","2024-01-13","2023-06-11","2023-05-02","2024-01-31",null,"2023-10-11","2024-02-01","2023-08-20","2023-10-25","2023-08-07","2023-09-29","2024-01-25","2023-10-20","2023-11-07","2023-12-02","2024-01-27","2024-02-01","2024-02-01","2023-09-06"]},"columns":[{"id":"name","name":"name","type":"character"},{"id":"topic","name":"topic","type":"character","minWidth":300},{"id":"purpose","name":"purpose","type":"character","minWidth":300},{"id":"no. members","name":"no. members","type":"numeric"},{"id":"created","name":"created","type":"Date"},{"id":"latest message","name":"latest message","type":"Date"}],"resizable":true,"searchable":true,"defaultSorted":[{"id":"name","desc":false}],"defaultPageSize":200,"outlined":true,"bordered":true,"striped":true,"dataKey":"678f400d65605e213e73126a1e7fb1ae"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>